tests/settings_tests/tests.py

```diff
@@ -441,3 +443,119 @@ def test_tuple_settings(self):
             finally:
                 del sys.modules['fake_settings_module']
                 delattr(settings_module, setting)
+
+
+class SettingChangeEnterException(Exception):
+    pass
+
+
+class SettingChangeExitException(Exception):
+    pass
+
+
+class OverrideSettingsIsolationOnExceptionTest(unittest.TestCase):
```

Comment:
Use `SimpleTestCase`.

Reply:
Sure, so I can use `self.addCleanup` or is there any other reason? (just curious)